### ISSUE TITLE
"QWidget::setMaximumSize: ... Negative sizes are not possible" when knitting with large patterns

### ISSUE THREAD

**Issue by chris007de** [AllYarnsAreBeautiful/ayab-desktop#417]
When knitting (for now, only tested with simulation) with a wide pattern (e.g. stirnband_160x20 or testPattern_200x40), the commandline is spammed with the following messages continuously:

```
QWidget::setMaximumSize: (/QWidget) Negative sizes (-1542979747,524287) are not possible
QWidget::setMaximumSize: (/QWidget) Negative sizes (-1542979747,524287) are not possible
QWidget::setMaximumSize: (/QWidget) Negative sizes (-1542979747,524287) are not possible
QWidget::setMaximumSize: (/QWidget) Negative sizes (-1542979747,524287) are not possible
QWidget::setMaximumSize: (/QWidget) Negative sizes (-1542979747,524287) are not possible
...
```

Happens with b1252f6292d8a3ecbee0096206769c0419aa479b

**Comment by t0mpr1c3** [comment]
My guess is that this happens when `KnitProgress.update()` tries to print outside the boundaries of the KnitProgress window.

It can probably be fixed by putting the function call that graphs each stitch inside a `try`... `except` structure.

It's definitely a bug and I'll get around to fixing this eventually.

**Comment by chris007de** [comment]
Seems to be fixed with #420
